marklogic/src/main/scala/quasar/physical/marklogic/fs/xdmitem.scala

```diff
@@ -21,52 +21,102 @@ import quasar.Data
+    case item: XdmText                  => Data._str(item.asString).point[F]
+    case item: XSAnyURI                 => Data._str(item.asString).point[F]
+    case item: XSBase64Binary           => bytesToData[F](item.asBinaryData)
+    case item: XSBoolean                => Data._bool(item.asPrimitiveBoolean).point[F]
+    case item: XSDate                   => Data.singletonObj("xs:date"      , Data.Str(item.asString)).point[F]
+    case item: XSDateTime               => Data._timestamp(Instant.ofEpochMilli(item.asDate.getTime)).point[F]
+    case item: XSDecimal                => Data._dec(item.asBigDecimal).point[F]
+    case item: XSDouble                 => Data._dec(item.asBigDecimal).point[F]
+    case item: XSDuration               => Data.singletonObj("xs:duration"  , Data.Str(item.asString)).point[F]
+    case item: XSFloat                  => Data._dec(item.asBigDecimal).point[F]
+    case item: XSGDay                   => Data.singletonObj("xs:gDay"      , Data.Str(item.asString)).point[F]
+    case item: XSGMonth                 => Data.singletonObj("xs:gMonth"    , Data.Str(item.asString)).point[F]
+    case item: XSGMonthDay              => Data.singletonObj("xs:gMonthDay" , Data.Str(item.asString)).point[F]
+    case item: XSGYear                  => Data.singletonObj("xs:gYear"     , Data.Str(item.asString)).point[F]
+    case item: XSGYearMonth             => Data.singletonObj("xs:gYearMonth", Data.Str(item.asString)).point[F]
+    case item: XSHexBinary              => bytesToData[F](item.asBinaryData)
+    case item: XSInteger                => Data._int(item.asBigInteger).point[F]
+    case item: XSQName                  => Data._str(item.asString).point[F]
+    case item: XSString                 => Data._str(item.asString).point[F]
+                                           // NB: This can be represented with org.threeten.bp.OffsetTime
+    case item: XSTime                   => Data.singletonObj("xs:time"      , Data.Str(item.asString)).point[F]
+    case item: XSUntypedAtomic          => Data._str(item.asString).point[F]
+    case other                          => s"No Data representation for '$other'.".wrapNel.raiseError[F, Data]
+  }
+
+  ////
+
+  private def bytesToData[F[_]: Applicative](bytes: Array[Byte]): F[Data] =
+    Data._binary(ImmutableArray.fromArray(bytes)).point[F]
+
+  private def jsonToData[F[_]: MonadErrMsgs](jsonString: String): F[Data] =
+    data.JsonParser.parseFromString(jsonString) match {
+      case Success(d) => d.point[F]
+      case Failure(e) => e.toString.wrapNel.raiseError[F, Data]
+    }
+
+  private def xmlToData[F[_]: MonadErrMsgs](xmlString: String): F[Data] = {
+    val el = SecureXML.loadString(xmlString).fold(_.toString.wrapNel.raiseError[F, Elem], _.point[F])
+
+    el flatMap { e =>
+      if (xml.qualifiedName(e) === xml.namespaces.ejsonEjson.shows)
+        data.fromXml[F](e)
+      else
+        xml.toData(e).point[F]
```

Comment:
Perhaps `decodeEJsonXml` or `decodeQuasarXml` just to make a clear distinction from `xml.toData`. I think `encodeXml` is fine but maybe making them symmetric is better. `decodeXml` is also not the end of the world.

Ah, I understand why we don't need to consider the case of some xml from our data in "normal" data. At first, I didn't realize we had a special encoding for everything, I thought maybe just the things that were harder to represent in xml.